client/loop.js

```diff
@@ -12,12 +12,17 @@ options.on({
 	'change:thumbs': reRenderImages,
 	'change:spoilers': toggleSpoilers,
 	'change:autogif': toggleAutoGIF,
-	'change:anonymise': toggleAnonymisation
+	'change:anonymise': toggleAnonymisation,
+	'change:workModeTOG': reRenderImages
 });
-
 function reRenderImages() {
-	follow(() => getImages((image, model) =>
-		image && model.dispatch('renderImage', image)));
+	if(main.state.page.get('catalog')){
+		//quick render, because we don't have models in the catalog
+		const show = (options.get("thumbs")!=='hide' && !options.get('workModeTOG'))? '':'none';
+		Array.prototype.forEach.call(document.querySelectorAll(".expanded"),(el, i)=> el.style.display=show);
```

Comment:
We are using https://github.com/WebReflection/dom4 polyfill to make sure all clients have the DOM level 4 spec available. So you can replace this with `document.queryAll(".expanded").forEach(el => el.style.display=show)`.